dpath/segments.py

```diff
@@ -34,7 +40,11 @@ def leafy(thing):
 
     leafy(thing) -> bool
     '''
-    return leaf(thing) or len(thing) == 0
+
+    try:
+        return leaf(thing) or len(thing) == 0
+    except:
+        return false
```

Comment:
1) Typo: this should be `False` not `false`. The fact that the tests pass mean this path isn't getting hit.
2) I recommend catching `TypeError` instead of a bare `except:`, as a bare `except` will also mask exceptions thrown by the interpreter due to syntax errors, block keyboard interrupts, etc.

Reply:
Good catch, the linter apparently needs adjusted. It was configured to catch these at one point.